src/Cake.Common/Tools/VSWhere/Product/VSWhereProduct.cs

```diff
@@ -55,6 +56,10 @@ private ProcessArgumentBuilder GetArguments(VSWhereProductSettings settings)
             {
                 builder.AppendQuoted(settings.Products);
             }
+            else
+            {
+                builder.AppendQuoted("*");
```

Comment:
This is not recommended. The default is Community, Professional, and Enterprise. Doing this you say "any product". The point was this is **vs**where and primarily geared for VS. If they want more, they should pass it in. Defaulting to "any product" really alters the behavior for anyone used to how vswhere.exe works.